go/teams/list.go

```diff
@@ -138,18 +154,31 @@ func fillUsernames(ctx context.Context, g *libkb.GlobalContext, res *keybase1.An
 		return err
 	}
 
+	for i, uid := range userList {
+		pkg := namePkgs[i]
+		for _, cb := range d.callbacks[uid] {
+			cb(pkg)
+		}
+	}
+
+	return nil
+}
+
+func fillUsernames(ctx context.Context, g *libkb.GlobalContext, res *keybase1.AnnotatedTeamList) error {
+	mapper := CreateDeferredUIDMapper()
+
 	for id := range res.Teams {
 		member := &res.Teams[id]
-		num := userSet[member.UserID]
-		pkg := namePkgs[num]
 
-		member.Username = pkg.NormalizedUsername.String()
-		if pkg.FullName != nil {
-			member.FullName = string(pkg.FullName.FullName)
-		}
+		mapper.MapUIDToPackage(member.UserID, func(pkg libkb.UsernamePackage) {
```

Comment:
cc @mlsteele I'm trying to figure out a way to use UIDMapper with cleaner code. Can you take a look?

The diff ended up a bit annoying (it greedily matched new helper functions with old `fillUsernames` body) so it might be best to look at the end results: `DeferredUIDMapper` and new `fillUsernames` function.